modules/platforms/dotnet/Apache.Ignite.Core.Tests/Cache/PartitionPreloadTest.cs

```diff
@@ -0,0 +1,294 @@
+
+        /** */
+        private const int GridCount = 3;
+
+        /** */
+        private const string MemCacheName = "mem-cache";
+
+        /** */
+        private const string PersistenceCacheName = "pds-cache";
+
+        /** */
+        private const int EntriesCount = 1000;
+
+        /// <summary>
+        /// Tests that preloading partition on client locally returns <code>false</code>.
+        /// </summary>
+        [Test]
+        public void TestLocalPreloadPartitionClient()
+        {          
+            var cache = GetClient.GetCache<int, int>(MemCacheName);
+            
+            Assert.IsFalse(cache.LocalPreloadPartition(0));
+        }
+        
+        /// <summary>
+        /// Preload partitions synchronously.
+        /// </summary>
+        [Test]
+        public void TestPreloadPartition()
+        {
+            PerformPreloadTest(GetGrid(0), GetClient, PreloadMode.Sync);
+        }
+
+        /// <summary>
+        /// Tests that incorrect usage of API throws exceptions.
+        /// </summary>
+        [Test]
+        public void TestPreloadPartitionInMemoryFail()
+        {
+            var cache = GetClient.GetCache<int, int>(MemCacheName);
+
+            Assert.Throws<CacheException>(() => cache.PreloadPartition(0));
+
+            var task = cache.PreloadPartitionAsync(0);
+
+            Assert.Throws<CacheException>(() => task.WaitResult());
+        }
+        
+        /// <summary>
+        /// Preload partitions asynchronously.
+        /// </summary>
+        [Test]
+        public void TestPreloadPartitionAsync()
+        {
+            PerformPreloadTest(GetGrid(0), GetClient, PreloadMode.Async);
+        }
+        
+        /// <summary>
+        /// Preload partitions locally.
+        /// </summary>
+        [Test]
+        public void TestLocalPreloadPartition()
+        {
+            PerformPreloadTest(GetGrid(0), GetClient, PreloadMode.Local);
+        }
+
+        /// <summary>
+        /// Test set up.
+        /// </summary>
+        [SetUp]
+        public void SetUp()
+        {
+            StartGrids();
+        }
+        
+        /// <summary>
+        /// Test tear down.
+        /// </summary>
+        [TearDown]
+        public void TearDown()
+        {
+            Ignition.StopAll(true);
+            
+            Directory.Delete(TempDir, true);
+        }
+        
+        /// <summary>
+        ///  Starts test grids.
+        /// </summary>
+        private static void StartGrids()
+        {
+            var cfg = new IgniteConfiguration(TestUtils.GetTestConfiguration())
+            {
+                DataStorageConfiguration = new DataStorageConfiguration()
+                {
+                    WalMode = WalMode.Fsync,
+                    WalSegmentSize = 16 * 1 << 20, // 16 MB.
+                    PageSize = 1 << 10, // 1 KB.
+                    MetricsEnabled = true,
+                    CheckpointFrequency = TimeSpan.FromSeconds(1),
```

Comment:
Please move `1` to some constant and use it below for the `Thread.Sleep` as well to make it clear where the delay comes from.